### PR TITLE
Added config.yml and README.md

### PR DESCRIPTION
**Pull request by Banbeucmas**
Allow configuration for switching and testing other value if needed. Keeping them inside the plugin class itself is not really that flexible.

Also added README.md. Check #2 for name suggestion ;)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# DangerousWater
+
+Allows water damage and more realistic water functionality. Originally intended for [Mine in Abyss](https://github.com/Derongan/MineInAbyss)
```
Comment (Derongan) [inline README.md]: I believe the current link is [Mine in Abyss](https://github.com/MineInAbyss)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+## Functionality
+
+* Damage users when falling into water
```
Comment (Derongan) [inline README.md]: Talk to @0ffz  a bit about this- I think there are a few intricacies worth putting in the readme

File: src/main/java/com/mineinabyss/enhancedwater/WaterListener.java [hunk 3]
```diff
@@ -13,10 +14,21 @@
...
-    double pushVelocity = 0.2; //velocity to push players off horizontally in waterfall
+    private double checkRange; //multiplier for blocks around player to be checked
+    private double fallRange; //block distance to check for player to be pushed while falling in water
+    private double pushVelocity; //velocity to push players off horizontally in waterfall
+
+    public WaterListener(EnhancedWater plugin) {
+        this.plugin = plugin;
+        this.config = plugin.getConfig();
+
+        checkRange = config.getDouble("checkRange");
```
Comment (Derongan) [inline src/main/java/com/mineinabyss/enhancedwater/WaterListener.java]: Nitpicky, but does this barf if the config is missing? Also if you saveDefaultConfig each time will it overwrite changes?

File: README.md (hunk 2 above)
Comment (Banbeucmas) [reply README.md]: @Derongan I might need more details on that, this section is based on my observation by looking at the source

File: src/main/java/com/mineinabyss/enhancedwater/WaterListener.java (hunk 3 above)
Comment (Banbeucmas) [reply src/main/java/com/mineinabyss/enhancedwater/WaterListener.java]: @Derongan [The plugin regenerates the config from source if it is not present, if you mean the file itself gets missing](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/plugin/Plugin.html#saveDefaultConfig--). This is fine except if we are going to implement a reload command.

But I don't deny the possibillity of an exception being thrown if the config node is missing. We can add some fallback value to the field.

**Pull request merged**